src/core/qgsarrayutils.cpp

```diff
@@ -0,0 +1,94 @@
+/***************************************************************************
+                                  qgsarrayutils.h
+                              ---------------------
+    begin                : July 2019
+    copyright            : (C) 2019 by David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsarrayutils.h"
+#include <QDebug>
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+QVariantList QgsArrayUtils::parse( const QString &string )
+{
+  QVariantList variantList;
+
+  QString newVal = string;
+  if ( newVal.trimmed().startsWith( '{' ) )
+  {
+    newVal =  newVal.trimmed().mid( 1 ).mid( 0, newVal.length() - 2 ).prepend( '[' ).append( ']' );
```

Comment:
You are right. Actually it cannot decode it with JSON because the "subarrays" are not quoted. I see what I can do. In the `qgspostgresprovider` we support multidimensional array,  but as far I understand it, we create still a one dimensional list containing "subarrays" as string, right?

Reply:
Yes, right. It could manage a real multidimensional array by calling recursively parseMultiDimensionnalArray but it's useless because there is no widget that know what to do with such data.